tuner/examples/test/tuner_test.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2024 Advanced Micro Devices, Inc
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""
+Sample Usage:
+
+python -m examples.test double_mmt.mlir mmt_benchmark.mlir --devices=hip://0,hip://1 --num-candidates=64
+
+
+Recommended Trial Run:
+
+python -m examples.test double_mmt.mlir mmt_benchmark.mlir --num-candidates=10
+
+
+Dry Run Test (no gpu required, not currently working):
+
+python -m examples.test double_mmt.mlir mmt_benchmark.mlir --num-candidates=64 --dry-run
+
+"""
+
+from tuner import libtuner
+from pathlib import Path, PurePath
+import argparse
+import os
+
+
+class TestTuner(libtuner.TuningClient):
+    def __init__(self):
+        self.compile_flags = [
+            "--iree-hip-target=gfx942",
```

Comment:
For dispatches, we don't need to provide the target as dispatches are already configured.

But we will need this for the full model,